components/confirm_modal.tsx

```diff
@@ -1,80 +1,83 @@
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 import {Modal} from 'react-bootstrap';
 import {FormattedMessage} from 'react-intl';
 
-export default class ConfirmModal extends React.Component {
-    static propTypes = {
-
-        /*
-         * Set to show modal
-         */
-        show: PropTypes.bool.isRequired,
-
-        /*
-         * Title to use for the modal
-         */
-        title: PropTypes.node,
-
-        /*
-         * Message to display in the body of the modal
-         */
-        message: PropTypes.node,
-
-        /*
-         * The CSS class to apply to the confirm button
-         */
-        confirmButtonClass: PropTypes.string,
-
-        /*
-         * The CSS class to apply to the modal
-         */
-        modalClass: PropTypes.string,
-
-        /*
-         * Text/jsx element on the confirm button
-         */
-        confirmButtonText: PropTypes.node,
-
-        /*
-         * Text/jsx element on the cancel button
-         */
-        cancelButtonText: PropTypes.node,
-
-        /*
-         * Set to show checkbox
-         */
-        showCheckbox: PropTypes.bool,
-
-        /*
-         * Text/jsx element to display with the checkbox
-         */
-        checkboxText: PropTypes.node,
-
-        /*
-         * Function called when the confirm button or ENTER is pressed. Passes `true` if the checkbox is checked
-         */
-        onConfirm: PropTypes.func.isRequired,
-
-        /*
-         * Function called when the cancel button is pressed or the modal is hidden. Passes `true` if the checkbox is checked
-         */
-        onCancel: PropTypes.func.isRequired,
-
-        /**
-         * Function called when modal is dismissed
-         */
-        onExited: PropTypes.func,
-
-        /*
-         * Set to hide the cancel button
-         */
-        hideCancel: PropTypes.bool,
-    }
+type Props = {
+
+    /*
+     * Set to show modal
+     */
+    show: boolean;
+
+    /*
+     * Title to use for the modal
+     */
+    title?: React.ReactNode;
+
+    /*
+     * Message to display in the body of the modal
+     */
+    message?: React.ReactNode;
+
+    /*
+     * The CSS class to apply to the confirm button
+     */
+    confirmButtonClass?: string;
+
+    /*
+     * The CSS class to apply to the modal
+     */
+    modalClass?: string;
+
+    /*
+     * Text/jsx element on the confirm button
+     */
+    confirmButtonText?: React.ReactNode;
+
+    /*
+     * Text/jsx element on the cancel button
+     */
+    cancelButtonText?: React.ReactNode;
+
+    /*
+     * Set to show checkbox
+     */
+    showCheckbox?: boolean;
+
+    /*
+     * Text/jsx element to display with the checkbox
+     */
+    checkboxText?: React.ReactNode;
+
+    /*
+     * Function called when the confirm button or ENTER is pressed. Passes `true` if the checkbox is checked
+     */
+    onConfirm: Function;
```

Comment:
Can we set the type for this one and the one below it to `(checked: boolean) => void`?
```suggestion
    onConfirm: (checked: boolean) => void;
```